ipld/retriever.go

```diff
@@ -56,13 +58,55 @@ func (r *Retriever) Retrieve(ctx context.Context, dah *da.DataAvailabilityHeader
 			if err == nil {
 				return eds, nil
 			}
+			if byzErr := r.handleByzantineError(ctx, dah, err); byzErr != nil {
+				return nil, byzErr
+			}
 		}
 		// retry quadrants until we can reconstruct the EDS or error out
 	}
 
 	return nil, format.ErrNotFound
 }
 
+func (r *Retriever) handleByzantineError(
```

Comment:
I would prefer if this were called `checkForByzantineError`

Reply:
Also can you add documentation to this method? And a couple comments explaining what you are doing inside so it's easier to follow.